scripts/cmake/vcpkg_configure_make.cmake

```
@@ -103,6 +103,17 @@ macro(z_vcpkg_extract_cpp_flags_and_set_cflags_and_cxxflags flag_suffix)
     string(STRIP "${CPPFLAGS_${flag_suffix}}" CPPFLAGS_${flag_suffix})
     string(STRIP "${CFLAGS_${flag_suffix}}" CFLAGS_${flag_suffix})
     string(STRIP "${CXXFLAGS_${flag_suffix}}" CXXFLAGS_${flag_suffix})
+
+    # Extract -isysroot from CFLAGS and add it to CPPFLAGS
+    # so that preprocessor-only configure tests get the right includes
+    separate_arguments(C_FLAGS NATIVE_COMMAND PROGRAM SEPARATE_ARGS "${VCPKG_DETECTED_CMAKE_C_COMPILER} ${VCPKG_DETECTED_CMAKE_C_FLAGS_${flag_suffix}}")

```

```suggestion
    separate_arguments(c_flags_sep NATIVE_COMMAND "${VCPKG_DETECTED_CMAKE_C_FLAGS_${flag_suffix}}")
```

Why do you need the special PROGRAM option with SEPARATE_ARGS here?

---

I may have been mislead by the examples on the docs page... Will try a simpler construct.